signmotion/flutter_image_converter
# web platform support

# Pull Request Template

## Status

READY

## Breaking Changes

YES

## Description
This PR removes the dependency of `dart:io` from the code.

I saw the item "Check out the Web platform" on your **TODO** list.  I've been working on the same for one of my published packages [google_vision_flutter](https://pub.dev/packages/google_vision_flutter), which has a dependency on this project.  I considered publishing the code in this PR as a `flutter_image_converter_lite` and use the lite version as a dependency to get web platform compliance, I'm instead submitting the code as a PR for you to decide what's best.  Until then, I'll incorporate just the code I need from this project into my own.

## Related PRs

List related PRs against other branches:

-

## Todos


## Steps to Test or Reproduce

I use the [pana](https://pub.dev/packages/pana) tool to confirm that the code when published to pub.dev will support the `web` platform.

```text
pana --no-dartdoc .
```

## Impact to Remaining Code Base

This PR will affect:
- this PR removed the dependency of `dart:io` and makes the package `web` platform compliant.  Any existing code that uses functionality tied to `dart:io` will no longer function.
- The code previously used to extend the `dart:io` `File` class has been moved into the test folder to ensure that all the existing test continue to run successfully.
